src/managed/SymbolReader.cs

```diff
@@ -767,12 +771,16 @@ SequencePoint FirstSequencePointForSourceLine(ref MetadataReader reader, int met
 
                         if (p.EndLine != nearestSP.EndLine)
                         {
-                            if (p.EndLine < nearestSP.EndLine)
+                            if (reqPos == Position.First && p.EndLine < nearestSP.EndLine)
+                                nearestSP = p;
+                            else if (reqPos == Position.Last && p.EndLine > nearestSP.EndLine)
```

Comment:
We probably don't need 2 branches for `if` statement here, just use 1 with modified condition:
```
    if ((a) ||
        (b))
         nearestSP = p;
```